lib/node_modules/@stdlib/array/float16/lib/polyfill.js

```diff
@@ -0,0 +1,538 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+class Float16Array {
+	constructor(input) {
+		if (typeof input === 'number') {
+			this._buffer = new Uint16Array(input);
+		} else if (Array.isArray(input)) {
+			this._buffer = new Uint16Array(input.length);
+			for (let i = 0; i < input.length; i++) {
+				this._buffer[i] = Float16Array.toFloat16(input[i]);
+			}
+		} else {
+			throw new TypeError('Unsupported constructor argument');
+		}
+		this.length = this._buffer.length;
+	}
+
+	get(index) {
+		return Float16Array.fromFloat16(this._buffer[index]);
+	}
+
+  get buffer() {
+	return this._buffer.buffer;
+  }
+
+  get byteLength() {
+	return this._buffer.byteLength;
+  }
+
+  get byteOffset() {
+	return this._buffer.byteOffset;
+  }
+
+  get BYTES_PER_ELEMENT() {
+    return 2;
+  }
+
+  get [Symbol.toStringTag]() {
+  return 'Float16Array';
+  }
+
+  copyWithin(target, start, end) {
+		target = target >> 0;
+		start = start >> 0;
+		end = end === undefined ? this.length : end >> 0;
+
+		const to = target < 0 ? this.length + target : target;
+		const from = start < 0 ? this.length + start : start;
+		const final = end < 0 ? this.length + end : end;
+
+		const count = Math.min(final - from, this.length - to);
+		const direction = from < to && to < (from + count) ? -1 : 1;
+
+		if (direction === -1) {
+			for (let i = count - 1; i >= 0; i--) {
+				this._buffer[to + i] = this._buffer[from + i];
+			}
+		} else {
+			for (let i = 0; i < count; i++) {
+				this._buffer[to + i] = this._buffer[from + i];
+			}
+		}
+		return this;
+	}
+
+  set(indexOrArray, value) {
+	if (typeof indexOrArray === 'number') {
+		if (indexOrArray < 0 || indexOrArray >= this.length) {
+			throw new RangeError('Index out of bounds');
+		}
+		this._buffer[indexOrArray] = Float16Array.toFloat16(value);
+		return;
+	}
+
+	const source = indexOrArray;
+	const offset = value !== undefined ? Number(value) : 0;
+
+	if (offset < 0 || offset > this.length) {
+		throw new RangeError('Offset is out of bounds');
+	}
+
+	if (
+		(Array.isArray(source) ||
+		(ArrayBuffer.isView(source) && typeof source.length === 'number') ||
+		(source instanceof Float16Array))
+	) {
+		const len = Math.min(source.length, this.length - offset);
```

Comment:
We do not use `Math` built-ins. Everything should be implemented in terms of stdlib functionality.